.infrastructure/identity/main.tf

```diff
@@ -122,3 +122,48 @@ resource "aws_iam_role_policy_attachment" "githubiac" {
   role       = aws_iam_role.githubiac.name
   policy_arn = data.aws_iam_policy.admin_access.arn
 }
+
+###############################################################################
+#                      Define IAM Role to use on deploy                       #
+###############################################################################
+data "aws_iam_policy" "deploy_access" {
+  name = "AmazonS3FullAccess"
+}
+
+resource "aws_iam_role" "githubaction_deploy" {
+  name        = "GitHubActionDeployRole"
+  description = "Role to assume to deploy the website"
+
+  assume_role_policy = jsonencode({
```

Comment:
The `assume_role_policy` is a special policy that defines which [principals](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html) can assume the role. I think that the provided `assume_role_policy` is incorrect. It should be the same as the one defined for `githubiac`.